### PR TITLE
dev-cmd/tap-new: use `Utils::Git.setup_gpg!`

### LINKED ISSUE #10992: `brew tap-new` fails if user's gitconfig has `commit.gpgsign true` and the GPG key requires a pinentry
### `brew config` output

```shell
HOMEBREW_VERSION: 3.0.10-80-g9fb771d
ORIGIN: https://github.com/Homebrew/brew
HEAD: 9fb771d2a4d9d6a3b0677abf6915b06a2767cddf
Last commit: 2 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 4e9060dd297d223a2066e02ea0f163b6bbbb4426
Core tap last commit: 44 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_DEVELOPER: set
HOMEBREW_EDITOR: nvim
HOMEBREW_GITHUB_API_TOKEN: set
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 12.0 build 1200
Git: 2.31.1 => /opt/homebrew/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 11.2.3-arm64
CLT: 12.4.0.0.1.1610135815
Xcode: 12.4
Rosetta 2: false
```

### `brew doctor` output

```shell
Your system is ready to brew.
```

### 

The `brew doctor` above contains no "Warning" lines.

### What were you trying to do (and why)?

I was trying to use `brew tap-new` to create a new tap. My git config happens to set `commit.gpgsign true`, and my private key is stored on a Yubikey smartcard (which was plugged in).

(I should add that manually committing from the command line works just fine)

### What happened (include all command output)?

```
$ brew tap-new --branch master apetresc/homebrew-magic
Initialized empty Git repository in /opt/homebrew/Library/Taps/apetresc/homebrew-magic/.git/
error: cannot run gpg: No such file or directory
error: gpg failed to sign the data
fatal: failed to write commit object
Error: Failure while executing; `git commit -m Create\ apetresc/magic\ tap` exited with 128.
```

Basically, the `brew tap-new` command failed halfway through. It complains that the problem is that it can't find `gpg` but I suspect the real root cause is that it's unable to run the pinentry program for some reason.

### What did you expect to happen?

I expect the same thing that happens if I run the failing command (`git commit -m Create\ apetresc/magic\ tap`) directly from the exact same CLI session: my pinentry program comes up, I enter the smartcard PIN, and it signs the commit for me.

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
1. Configure your git according to best practices, with `commit.gpgsign true` and private keys stored on a hardware device such as a Yubikey smartcard, with a PIN.
2. Attempt to run `brew tap-new`.
```

### PR DESCRIPTION
**Pull request by nandahkrishna**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

This PR fixes #10992. We should `setup_gpg!` before committing.

### CODE REVIEW

**Comment by nandahkrishna** [comment]
I checked the tests for the other `dev-cmd`s that use `Utils::Git.setup_gpg!` and it seems like that part isn't tested/run. I didn't want to remove the test for `tap-new`, so I've just setup `gnupg` as a test formula.

File: Library/Homebrew/test/dev-cmd/tap-new_spec.rb [hunk 1]
```diff
@@ -7,6 +7,13 @@
   it_behaves_like "parseable arguments"
 
   it "initializes a new tap with a README file and GitHub Actions CI", :integration_test do
+    # To ensure that Utils::Git.setup_gpg! doesn't raise an error
+    setup_test_formula "gnupg", <<~RUBY
+      desc "GNU Pretty Good Privacy (PGP) package"
+      homepage "https://gnupg.org"
+      url "https://gnupg.org/ftp/gcrypt/gnupg/gnupg-2.2.27.tar.bz2"
+    RUBY
```
Comment (MikeMcQuaid) [inline Library/Homebrew/test/dev-cmd/tap-new_spec.rb:15]:
```suggestion
    setup_test_formula "gnupg"
```

If this works I think it's preferable.

**Review by MikeMcQuaid — APPROVED** [review]
Looks good, nice work @nandahkrishna!

File: Library/Homebrew/test/dev-cmd/tap-new_spec.rb (hunk 1 above)
Comment (nandahkrishna) [reply Library/Homebrew/test/dev-cmd/tap-new_spec.rb]: Done, I had to modify `integration_test.rb` to get it to work, similar to `patchelf`.

**Review by MikeMcQuaid — APPROVED** [review]

**Pull request merged**